### PR TITLE
feat: parse readme and get templates from zip

### PR DESCRIPTION
**Pull request by RonenMars**
## Description
Fetch templates from github in the runtime and display in the UI:

1. Get the dist.zip from kittehub
2. Unpack and parse the zip
3. Filter all relevant README files with metadata
4. Put the whole information in local-storage except the files
5. Put the files in the indexedDB
6. Add check if kittehub had newer last commit
7. Run this check ^ maximum every day

## Linear Ticket
https://linear.app/autokitteh/issue/UI-898/fetch-templates-from-github-in-the-runtime-and-display-in-the-ui

## What type of PR is this? (check all applicable)

-   [x] 💡 (feat) - A new feature (non-breaking change which adds functionality)
-   [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
-   [ ] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
-   [ ] 🏎 (perf) - Optimization
-   [ ] 📄 (docs) - Documentation - Documentation only changes
-   [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
-   [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] ☑️ (chore) - Chores - Other changes that don't modify src or test files
-   [ ] ↩️ (revert) - Reverts - Reverts a previous commit(s).

### CODE REVIEW

File: src/hooks/useCreateProjectFromTemplate.tsx [hunk 1]
```diff
@@ -115,5 +111,24 @@ export const useCreateProjectFromTemplate = () => {
 		}
 	};
 
-	return { createProjectFromTemplate };
+	const createProjectFromAsset = async (templateAssetDirectory: string, projectName?: string) => {
+		setIsCreating(true);
+		try {
+			const template = findTemplateByAssetDirectory(templateAssetDirectory);
+			if (!template) {
+				throw new Error(`Template not found for asset directory: ${templateAssetDirectory}`);
```
Comment (J1za) [inline src/hooks/useCreateProjectFromTemplate.tsx:119]: add to i18n

File: src/hooks/useCreateProjectFromTemplate.tsx [hunk 2]
```diff
@@ -115,5 +111,24 @@ export const useCreateProjectFromTemplate = () => {
...
+			}
+			await createProjectFromTemplate(template, projectName);
+		} catch (error) {
+			addToast({
+				message: t("projectCreationFailed"),
+				type: "error",
+			});
+			console.error("Error creating project from template:", error);
```
Comment (J1za) [inline src/hooks/useCreateProjectFromTemplate.tsx:127]: Why is here console.error?

File: src/utilities/fetchAndExtractZip.utils.ts [hunk 3]
```diff
@@ -243,7 +214,7 @@ export const processReadmeFiles = (fileStructure?: FileStructure | null): Proces
 						? attributes.categories
 						: [attributes.categories];
 
-					categories.forEach((category) => {
+					categories.forEach((category: string) => {
```
[the remark this answers is not in the document]
Comment (RonenMars) [reply src/utilities/fetchAndExtractZip.utils.ts:217]: fixed

File: src/hooks/useCreateProjectFromTemplate.tsx (hunk 1 above)
Comment (RonenMars) [reply src/hooks/useCreateProjectFromTemplate.tsx:119]: done

File: src/hooks/useCreateProjectFromTemplate.tsx (hunk 2 above)
Comment (RonenMars) [reply src/hooks/useCreateProjectFromTemplate.tsx:127]: done

File: src/store/useTemplatesStore.ts [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+					ns: "stores",
+					error: error?.response?.data,
+				});
+			} else {
+				logErrorMessage = i18n.t("templates.failedToFetchExtended", {
+					ns: "stores",
+					error: error?.message,
+				});
+			}
+			console.error(logErrorMessage);
```
Comment (J1za) [inline src/store/useTemplatesStore.ts:171]: Why is not Logger?

File: src/store/useTemplatesStore.ts [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+			}
+
+			if (!shouldFetchTemplates) {
+				set({ isLoading: false });
+
+				return;
+			}
+
+			if (shouldFetchTemplatesFromGithub) {
+				const { categories, templateMap } = await processTemplates(remoteTemplatesArchiveURL);
```
[the remark this answers is not in the document]
Comment (RonenMars) [reply src/store/useTemplatesStore.ts]: refactored

File: src/store/useTemplatesStore.ts [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+			const sortedCategories = sortCategories(categories, templateCategoriesOrder);
+			set({
+				templateMap,
+				sortedCategories,
+				lastCommitDate,
+				isLoading: false,
+				error: null,
+			});
+
+			return;
```
[the remark this answers is not in the document]
Comment (RonenMars) [reply src/store/useTemplatesStore.ts]: done

[the remark this answers is not in the document]
Comment (RonenMars) [reply src/store/useTemplatesStore.ts]: done

File: src/store/useTemplatesStore.ts [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+				throw new Error(result.error);
+			}
+
+			const processedCategories: ProcessedCategory[] = processReadmeFiles(result.structure);
+			const templateMap: Record<string, TemplateMetadataWithCategory> = {};
+			const { templateStorage } = get();
+
+			await Promise.all(
+				processedCategories.map(async (category) => {
+					await Promise.all(
```
[the remark this answers is not in the document]
Comment (RonenMars) [reply src/store/useTemplatesStore.ts]: done

File: src/interfaces/store/templates.interface.ts [hunk 8]
```diff
@@ -1,27 +1,73 @@
...
+
+export interface TemplateState {
+	templateMap: Record<string, TemplateMetadata>;
+	isLoading: boolean;
+	sortedCategories?: TemplateCategory[];
+	error: string | null;
+	lastCommitDate?: string;
+	lastCheckDate?: Date;
+	templateStorage: TemplateStorageService;
+
```
[the remark this answers is not in the document]
Comment (RonenMars) [reply src/interfaces/store/templates.interface.ts:69]: removed

File: src/locales/en/dashboard/translation.json [hunk 9]
```diff
@@ -20,7 +20,10 @@
 		"projectTemplateFilesNotFound": "Project template files not found",
 		"projectFilesNotFound": "Project files not found",
 		"projectDirectoryNotConfigured": "Project directory not configured",
-		"projectFilesNotFoundExtended": "Project files not found, error: {{error}}"
+		"projectFilesNotFoundExtended": "Project files not found, error: {{error}}",
```
[the remark this answers is not in the document]
Comment (RonenMars) [reply src/locales/en/dashboard/translation.json:23]: removed and removed a few more

File: src/store/useTemplatesStore.ts (hunk 4 above)
Comment (RonenMars) [reply src/store/useTemplatesStore.ts:171]: Basically we need only console.
But in logger I added parameter `consoleOnly` and passed `true` now.

**Review by J1za — APPROVED** [review]

File: src/components/organisms/dashboard/templates/tabs/tabs.tsx [hunk 10]
```diff
@@ -1,72 +1,80 @@
...
+	const [selectedTemplate, setSelectedTemplate] = useState<TemplateMetadata>();
+
 	const { openModal } = useModalStore();
+	const { error, fetchTemplates, isLoading, sortedCategories: categories } = useTemplatesStore();
 
 	const activeCategory = useMemo(
-		() => templateProjectsCategories.find((category) => category.name === activeTab),
-		[activeTab]
+		() => categories?.find((category) => category.name === activeTab),
+		[activeTab, categories]
```
[the remark this answers is not in the document]
Comment (RonenMars) [reply src/components/organisms/dashboard/templates/tabs/tabs.tsx:20]: We need it to refresh the view when the categories change.

**Pull request merged**